packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -208,7 +208,7 @@ - (NSDictionary *)encodingMap
                     resolver:(EXPromiseResolveBlock)resolve
                     rejecter:(EXPromiseRejectBlock)reject)
 {
-  NSURL *uri = [NSURL URLWithString:uriString];
+  NSURL *uri = [NSURL URLWithString:[uriString stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]]];
```

Comment:
At a glance, we should not be double-encoding URIs. Something seems off here. Maybe the parameter needs to be renamed to something other than uriString because if it were a valid URI, it should have been encoded in the first place.

Reply:
I believe we're not double-encoding URIs here as `uriString` is a `NSString *`, but I agree that `uriString`  may be misleading

Reply:
I've renamed this to `pathString` instead.